src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,122 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/serializer/Uuid.h"
+
+#include <arpa/inet.h>
+
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::serializer::uuid {
+
+std::string AsString::serialize(const uprotocol::v1::UUID uuid) {
+	// Extracting the parts of the UUIDv8
+	uint64_t unix_ts_ms = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;  // 48 bit
+	uint8_t ver = (uuid.msb() >> 12) & 0xF;                     // 4 bit
+	uint16_t counter = uuid.msb() & 0xFFF;                      // 12 bit
+	uint8_t var = (uuid.lsb() >> 62) & 0x3;                     // 2 bit
+	uint64_t rand_b = uuid.lsb() & 0x3FFFFFFFFFFFFFFF;          // 62 bit
```

Comment:
Lots of shifting and bitmasking -- maybe more constexpr would help make it clearer ?  The UUID validator cleaned up amazing well, I'm hoping for the same(ish).